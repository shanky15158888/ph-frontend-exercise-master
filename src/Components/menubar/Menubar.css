nav {
    min-height: 8vh;
  }
  li{
    list-style-type: none;
  }
  .nav-links a{
    color: black
  }
  .burger {
    cursor:pointer;
    margin-top: 22px;
  }
  .burger div{
    width: 25px;
    height: 2px;
    background-color: #7f187f;
    margin: 5px;
  }
  .nav-links{
    position: absolute;
    left: 0px;
    height: 92vh;
    top: 5vh;
    border-right: 2px solid #dfdfe1;
  }
  .left-menubar {
    border-right: 2px solid #dfdfe1;
    margin-left: 15px;
  }
  ul ul {
    padding-left: 14px;
  }
  .navbar-scroll {
    margin-top: 27px;
  }
  .navbar-scroll > ul {
    padding-left: 11px;
  }
  a {
    margin-left:10px;
    font-size: 14px;
    color: #211e22;
  }
  a:hover {
    text-decoration: none;
  }
  .dn {
    display: none;
  }
  .arrow {
    float: right;
    margin: 4px 20px;
  }
  .child {
    transition: max-height 0.5s ease-in-out;
    overflow: hidden;
    max-height: 0;
  }
  .active {
    max-height: 1000px;
  }
  ul {
    width: 100%;
  }
    