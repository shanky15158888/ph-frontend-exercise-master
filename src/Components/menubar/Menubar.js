import React from 'react'
import Lists from './Lists'
import './Menubar.css';

function Menubar({ updateWidth, fetchData, data }) {
    const { width, initialNavBarWidth, expandedNavBarWidth, navData } = data;
    return (
        <div className="left-menubar" style={{ width: `${width}%` }}>
            <div className="burger" onClick={() => updateWidth(width === expandedNavBarWidth ? initialNavBarWidth : expandedNavBarWidth)}>
                <div className="line1"></div>
                <div className="line2"></div>
                <div className="line3"></div>
            </div>
            <div className="navbar-scroll">
                <Lists
                    showTitle={width === initialNavBarWidth ? false : true}
                    data={navData} fetchData={fetchData}
                />
            </div>
        </div>
    )
}
export default Menubar
