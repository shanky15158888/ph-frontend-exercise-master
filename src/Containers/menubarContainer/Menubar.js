import React from 'react'
import { connect } from 'react-redux'
import Menubar from '../../Components/menubar/Menubar'
import { navbarWidthAction, thunk_action_creator } from '../../actions/actions'

const mapStateToProps = (state, ownProps) => {
    return {
        data: state.MenubarReducer
    }
}

const mapDispatchToProps = (dispatch, ownProps) => {
    return {
        updateWidth: (param) => dispatch(navbarWidthAction(param)),
        fetchData: (url) => dispatch(thunk_action_creator(url))
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(Menubar)
