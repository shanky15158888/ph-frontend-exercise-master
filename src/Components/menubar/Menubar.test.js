import React from 'react';
import renderer from 'react-test-renderer';
import Menubar from './Menubar';

test('Menubar testing', () => {
    const json = {
        data: {
            width: "3",
            initialNavBarWidth: "3",
            expandedNavBarWidth: "18",
            updateWidth: () => { },
            fetchData: () => { },
            navData: [
                {
                    title: 'Reporting',
                    url: null,
                    id: 'menu-reporting',
                    icon: 'reporting',
                    hasAlert: null,
                    children: [
                        {
                            title: 'Sales Dashboard',
                            url: 'https://partnerswayfaircom.csnzoo.com/v/sales_dashboard/index',
                            id: 'menu-sales-dashboard',
                            icon: null,
                            hasAlert: null,
                            children: []
                        },
                        {
                            title: 'Customer Incidents Dashboard',
                            url: 'https://partnerswayfaircom.csnzoo.com/v/customer-incidents-dashboard/index',
                            id: 'menu-customer-incidents-dashboard',
                            icon: null,
                            hasAlert: null,
                            children: []
                        }
                    ]
                },
                {
                    title: 'Tickets',
                    url: 'https://partnerswayfaircom.csnzoo.com/app/tickets',
                    id: 'menu-ticket',
                    icon: 'tickets',
                    hasAlert: null,
                    children: []
                }
            ]
        }
    }
    const component = renderer.create(
        <Menubar {...json} />,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});