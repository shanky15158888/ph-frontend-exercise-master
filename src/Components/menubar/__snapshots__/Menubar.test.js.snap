// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menubar testing 1`] = `
<div
  className="left-menubar"
  style={
    Object {
      "width": "3%",
    }
  }
>
  <div
    className="burger"
    onClick={[Function]}
  >
    <div
      className="line1"
    />
    <div
      className="line2"
    />
    <div
      className="line3"
    />
  </div>
  <div
    className="navbar-scroll"
  >
    <ul>
      <li>
        <span>
          <i
            className="bi bi-clipboard-data"
          />
        </span>
        <ul
          className="child "
        >
          <li />
          <li />
        </ul>
      </li>
      <li>
        <span>
          <i
            className="bi bi-clipboard-data"
          />
        </span>
      </li>
    </ul>
  </div>
</div>
`;
